src/ServiceInsight/ServiceControl/DefaultServiceControl.cs

```diff
@@ -533,4 +532,14 @@ public PresentationHint Hint
             set { hint = value; }
         }
     }
+
+    public class ResponseUnicodeCleaner
```

Comment:
Without knowing too much about the `DefaultServiceControl` class, my instinct would be to have a private method on `DefaultServiceControl` to do this work rather than a new class.

From my naive POV `DefaultServiceControl` does seem to be doing a lot but if I were to break it apart I'd probably start top down rather than bottom up.